app/client/src/components/shared/MapWidgets.tsx

```diff
@@ -1378,6 +1504,215 @@ function ShowAllWaterbodies({
+        abortController.abort();
+
+        setSurroundingMonitoringLocations({
+          status: 'fetching',
+          data: null,
+        });
+        abortController = new AbortController();
+
+        // convert the extent into northwest and southeast corner points
+        const northwestWM = new Point({
+          x: mapView.extent.xmin,
+          y: mapView.extent.ymax,
+        });
+        const southwestWM = new Point({
+          x: mapView.extent.xmax,
+          y: mapView.extent.ymin,
+        });
+
+        // convert the points to geographic
+        const northwest = webMercatorUtils.webMercatorToGeographic(
+          northwestWM,
+          false,
+        ) as __esri.Point;
+        const southwest = webMercatorUtils.webMercatorToGeographic(
+          southwestWM,
+          false,
+        ) as __esri.Point;
+
+        // get the bbox values from the northwest and southeast corner points
+        const north = northwest.latitude;
+        const south = southwest.latitude;
+        const east = southwest.longitude;
+        const west = northwest.longitude;
+
+        const url =
+          services.data.waterQualityPortal.monitoringLocation +
+          `search?mimeType=geojson&zip=no&bBox=${west},${south},${east},${north}`;
+        fetchCheck(url, abortController.signal)
+          .then((res: MonitoringLocationsData) => {
+            const idsToFilterOut: string[] = [];
+            const monitoringLocations = getMonitoringLocations();
+
+            if (monitoringLocations.status === 'success') {
+              monitoringLocations.data.features.forEach((location) => {
+                idsToFilterOut.push(
+                  location.properties.MonitoringLocationIdentifier,
+                );
+              });
+            }
+
+            const newData: FetchState<MonitoringLocationsData> = {
+              status: 'success',
+              data: {
+                ...res,
+                features: res.features.filter(
+                  (location) =>
+                    !idsToFilterOut.includes(
+                      location.properties.MonitoringLocationIdentifier,
+                    ),
+                ),
+              },
+            };
+            setSurroundingMonitoringLocations(newData);
+
+            const stations = buildStations(
+              newData,
+              surroundingMonitoringLocationsLayer,
+            );
+            if (!stations) return;
+
+            updateMonitoringLocationsLayer(
+              stations,
+              surroundingMonitoringLocationsLayer,
+            );
+          })
+          .catch((err) => {
+            if (isAbort(err)) return;
+            console.error(err);
+            setSurroundingMonitoringLocations({
+              status: 'failure',
+              data: null,
+            });
+          });
+      },
+    );
+
+    // watch for zoom changes and hide the surroundingMonitoringLocationsLayer when
+    // zoomed out to much
```

Comment:
Just a small comment typo:
```suggestion
    // zoomed out too much
```